File: symfony/src/Controller/HomeController.php
<?php

namespace App\Controller;

use App\Entity\SprintItem;
use App\Repository\SprintItemRepository;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\HttpKernel\Kernel;
use Symfony\Component\Routing\Attribute\Route;

class HomeController extends AbstractController
{
    public function __construct(private readonly SprintItemRepository $repository)
    {
    }

    #[Route('/', name: 'home', methods: ['GET'])]
    public function index(): Response
    {
        // Sprint 2: los items salen de MySQL (Doctrine), no de un array hardcodeado.
        // El cálculo y el contrato visible no cambian respecto al Sprint 1.
        $items = $this->repository->findAllOrdered();
        $totalWeight = array_sum(array_map(fn (SprintItem $i) => $i->getWeight(), $items));
        $doneWeight = array_sum(array_map(
            fn (SprintItem $i) => 'done' === $i->getStatus() ? $i->getWeight() : 0,
            $items
        ));
        $completion = 0 === $totalWeight ? 0 : intdiv($doneWeight * 100, $totalWeight);

        return $this->render('home/index.html.twig', [
            'items' => $items,
            'totalItems' => count($items),
            'totalWeight' => $totalWeight,
            'completion' => $completion,
            'framework' => 'Symfony',
            'frameworkVersion' => Kernel::VERSION,
        ]);
    }
}

File: springboot/src/main/java/com/sincrodev/sidebyside/web/HomeController.java
package com.sincrodev.sidebyside.web;

import com.sincrodev.sidebyside.domain.SprintItem;
import com.sincrodev.sidebyside.domain.SprintItemRepository;
import org.springframework.boot.SpringBootVersion;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

import java.util.List;

@Controller
public class HomeController {

    private final SprintItemRepository repository;

    public HomeController(SprintItemRepository repository) {
        this.repository = repository;
    }

    @GetMapping("/")
    public String home(Model model) {
        // Sprint 2: los items salen de MySQL (Spring Data JPA), no de un List hardcodeado.
        // El cálculo y el contrato visible no cambian respecto al Sprint 1.
        List<SprintItem> items = repository.findAllByOrderByIdAsc();

        int totalWeight = items.stream().mapToInt(SprintItem::getWeight).sum();
        int doneWeight = items.stream()
            .filter(i -> "done".equals(i.getStatus()))
            .mapToInt(SprintItem::getWeight)
            .sum();
        int completion = totalWeight == 0 ? 0 : (doneWeight * 100) / totalWeight;

        model.addAttribute("items", items);
        model.addAttribute("totalItems", items.size());
        model.addAttribute("totalWeight", totalWeight);
        model.addAttribute("completion", completion);
        model.addAttribute("framework", "Spring Boot");
        model.addAttribute("frameworkVersion", SpringBootVersion.getVersion());

        return "index";
    }
}
